RecipeBook/Data/UserDefaultsService.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  UserDefaultsName.swift
+//  RecipeBook
+//
+//  Created by David Farcas on 10.08.2023.
+//
+import Foundation
+
+
+class UserDefaultsService {
+
+    let userDefaults = UserDefaults.standard
+
+    func setUserName(_ name: String) {
+        userDefaults.set(name, forKey: UserDefaultsKeys.username.rawValue)
+    }
+
+    func getUserName() -> String? {
+        let username = userDefaults.string(forKey: UserDefaultsKeys.username.rawValue)
+        return username ?? ""
+    }
+
+    // Check first launch
+    func isFirstLaunch() -> Bool {
```

Comment:
I think this logic is not necessary. It's enough to only create a getter and a setter for one key in here and from a view model use these methods to determine whether or not the app is launched for the first time.